src/plugins/discover/public/application/main/hooks/utils/build_state_subscribe.ts

```diff
@@ -50,14 +50,33 @@ export const buildStateSubscribe =
       return;
     }
     addLog('[appstate] subscribe triggered', nextState);
-    const { hideChart, interval, breakdownField, sort, index } = appState.getPrevious();
+    const {
+      hideChart,
+      interval,
+      breakdownField,
+      sort,
+      index,
+      query: prevQuery,
+    } = appState.getPrevious();
+    const nextQuery = nextState.query;
+
+    const isTextBasedQueryLang = isTextBasedQuery(nextQuery);
+
+    if (isTextBasedQueryLang) {
+      const isTextBasedQueryLandPrev = isTextBasedQuery(prevQuery);
+      if (!isTextBasedQueryLandPrev) {
+        savedSearchState.update({ nextState });
+        dataState.reset(savedSearch);
+      }
+    }
     // Cast to boolean to avoid false positives when comparing
     // undefined and false, which would trigger a refetch
     const chartDisplayChanged = Boolean(nextState.hideChart) !== Boolean(hideChart);
-    const chartIntervalChanged = nextState.interval !== interval;
+    const chartIntervalChanged = nextState.interval !== interval && !isTextBasedQueryLang;
     const breakdownFieldChanged = nextState.breakdownField !== breakdownField;
-    const docTableSortChanged = !isEqual(nextState.sort, sort);
-    const dataViewChanged = !isEqual(nextState.index, index);
+    const docTableSortChanged = !isEqual(nextState.sort, sort) && !isTextBasedQueryLang;
+    const dataViewChanged = !isEqual(nextState.index, index) && !isTextBasedQueryLang;
+    const queryChanged = !isEqual(nextQuery, prevQuery);
```

Comment:
The change of query is now only handled here, no more observables can trigger refetches